tensorflow/lite/micro/kernels/reduce_common.cc

```diff
@@ -308,4 +308,68 @@ TfLiteStatus EvalMaxHelper(TfLiteContext* context, TfLiteNode* node,
   return kTfLiteOk;
 }
 
+TfLiteStatus EvalSumHelper(TfLiteContext* context, TfLiteNode* node,
+                           OpDataReduce* op_data) {
+  const TfLiteEvalTensor* input = tflite::micro::GetEvalInput(context, node, 0);
+  const TfLiteEvalTensor* axis = tflite::micro::GetEvalInput(context, node, 1);
+  TfLiteEvalTensor* output = tflite::micro::GetEvalOutput(context, node, 0);
+  TF_LITE_ENSURE_TYPES_EQ(context, input->type, output->type);
+  TfLiteReducerParams* params =
+      static_cast<TfLiteReducerParams*>(node->builtin_data);
+
+  // Interpret an axis tensor with null dimensions as a scalar.
+  int num_axis = static_cast<int>(ElementCount(*axis->dims));
+  int temp_index[kMaxNumberOfAxis];
+  int resolved_axis[kMaxNumberOfReducedAxis];
+
+  switch (input->type) {
+    case kTfLiteFloat32: {
+      TF_LITE_ENSURE(
+          context,
+          reference_ops::ReduceGeneric<float>(
+              tflite::micro::GetTensorData<float>(input), input->dims->data,
+              input->dims->size, tflite::micro::GetTensorData<float>(output),
+              output->dims->data, output->dims->size,
+              tflite::micro::GetTensorData<int>(axis), num_axis,
+              params->keep_dims, temp_index, resolved_axis, /*init_value=*/0.f,
+              [](const float current, const float in) -> float {
+                return in + current;
+              }));
+    } break;
+    case kTfLiteInt8: {
+      int32_t* temp_buffer = static_cast<int32_t*>(
+          context->GetScratchBuffer(context, op_data->temp_buffer_idx));
+      TF_LITE_ENSURE(
+          context,
+          reference_ops::QuantizedMeanOrSum(
+              tflite::micro::GetTensorData<int8_t>(input), op_data->input_zp,
+              op_data->input_scale, input->dims->data, input->dims->size,
+              tflite::micro::GetTensorData<int8_t>(output), op_data->output_zp,
+              op_data->output_scale, output->dims->data, output->dims->size,
+              tflite::micro::GetTensorData<int>(axis), num_axis,
+              params->keep_dims, temp_index, resolved_axis, temp_buffer,
+              /*compute_sum=*/true));
+    } break;
+    case kTfLiteInt16: {
+      int32_t* temp_buffer = static_cast<int32_t*>(
+          context->GetScratchBuffer(context, op_data->temp_buffer_idx));
+      TF_LITE_ENSURE(
+          context,
+          reference_ops::QuantizedMeanOrSum(
+              tflite::micro::GetTensorData<int16_t>(input), op_data->input_zp,
+              op_data->input_scale, input->dims->data, input->dims->size,
+              tflite::micro::GetTensorData<int16_t>(output), op_data->output_zp,
+              op_data->output_scale, output->dims->data, output->dims->size,
+              tflite::micro::GetTensorData<int>(axis), num_axis,
+              params->keep_dims, temp_index, resolved_axis, temp_buffer,
+              /*compute_sum=*/true));
+    } break;
+    default:
+      TF_LITE_ENSURE_MSG(context, false,
```

Comment:
Could you change this to using `MicroPrintf` and returning an error instead? See the default case in `EvalMaxHelper`.

Reply:
Done in commit db07f0ff96e5476de178bec59e463ac1b437ec6c